pyproject.toml

```
@@ -0,0 +1,123 @@
+[project]
+name = "zppy_interfaces"
+dynamic = ["version"]
+authors = [
+    { name="Ryan Forsyth", email="forsyth2@llnl.gov" }
+]
+description = "A package for providing extra functionality on top of external packages"
+license = {file = "LICENSE"}
+readme = "README.md"
+requires-python = ">=3.9"
+classifiers = [
+    # these are only for searching/browsing projects on PyPI
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: 3.12",
+    "Programming Language :: Python :: 3.13",
+
+    "License :: OSI Approved :: BSD License",
+    "Operating System :: OS Independent",
+]
+
+dependencies = [
+    "matplotlib >=3.8.2,<3.10",
+    "netcdf4",
+    "numpy >=2.0,<3.0",
+    "xarray >=2023.02.0",
+    "xcdat >=0.7.3,<1.0",
+]
+
+[project.optional-dependencies]
+
+testing = [
+    "pytest",
+    "pytest-cov",
+]
+
+docs = [
+    "sphinx",
+    "sphinx_rtd_theme",
+    "sphinx-multiversion",
+]
+
+qa = [
+    "black==23.9.1",
+    "flake8==6.1.0",
+    "flake8-isort==6.1.0",
+    "isort==5.12.0",
+    "mypy==1.5.1",
+    "pre-commit >=3.0.0",
+    "types-PyYAML >=6.0.0",
+]
```

```suggestion
qa = [
    "black==24.10.1",
    "flake8==7.1.1",
    "flake8-isort==6.1.0",
    "isort==5.13.2",
    "mypy==1.11.2",
    "pre-commit >=3.0.0",
    "types-PyYAML >=6.0.0",
]
```

Versions should align with `.pre-commit-config.yaml`.